### PR TITLE
FEATURE: Add detection of post date to use in bookmark modal

### PR DESCRIPTION
**Pull request by martin-brennan**
This PR introduces a feature that will detect a date inside the post that a user is bookmarking, and offer that date as an option in the bookmark modal.

image

image

The logic is that we get the first date/time detected in the post. If it does not have a time, just a date, then we default to 8:00am for the time.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/models/post.js [hunk 1]
```diff
@@ -305,6 +305,12 @@ const Post = RestModel.extend({
   },
 
   toggleBookmark() {
+    let postEl = document.querySelector(`[data-post-id="${this.id}"]`);
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/models/post.js:308]: Not convinced that this is the cleanest way to do this...

File: app/assets/javascripts/discourse/app/models/post.js [hunk 2]
```diff
@@ -313,6 +319,12 @@ const Post = RestModel.extend({
           reminderAt: this.bookmark_reminder_at,
           autoDeletePreference: this.bookmark_auto_delete_preference,
           name: this.bookmark_name,
+          postDetectedLocalDate: localDateEl
+            ? localDateEl.getAttribute("data-date")
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/models/post.js:323]: better use dataset

File: app/assets/javascripts/discourse/app/models/post.js [hunk 3]
```diff
@@ -313,6 +319,12 @@ const Post = RestModel.extend({
...
+            : null,
+          postDetectedLocalTime: localDateEl
+            ? localDateEl.getAttribute("data-time")
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/models/post.js:326]: better use dataset

File: app/assets/javascripts/discourse/app/models/post.js (hunk 1 above)
Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/models/post.js:308]: doesn't strike me as being terrible:) this is a simple and pragmatic solution...

Future might prove it ot be brittle, but we can advise if it's the case

**Comment by martin-brennan** [comment]
Thanks @jjaffeux, I always forget about dataset because it is "new"

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**